lxd/cluster/recover.go

```diff
@@ -507,6 +507,31 @@ func unpackTarball(tarballPath string, destRoot string) error {
 	return nil
 }
 
+func createDatabaseBackup(databaseDir string) (string, error) {
+	varDir := path.Dir(databaseDir)
+
+	// tar interprets `:` as a remote drive; ISO8601 allows a 'basic format'
+	// with the colons omitted (as opposed to time.RFC3339)
+	// https://en.wikipedia.org/wiki/ISO_8601
+	backupFileName := fmt.Sprintf("db_backup.%s.tar.gz", time.Now().Format("2006-01-02T150405Z0700"))
+	tarballPath := filepath.Join(varDir, backupFileName)
+
+	walkDir := path.Base(databaseDir)
+
+	logger.Info("Creating database backup", logger.Ctx{"archive": tarballPath})
+
+	// Don't include the recovery tarball in a backup tarball
+	excludeFiles := []string{path.Join(walkDir, RecoveryTarballName)}
+
+	err := createTarball(tarballPath, varDir, walkDir, excludeFiles)
+	if err != nil {
+		_ = os.Remove(tarballPath)
```

Comment:
this should be the responsibility of createTarball otherwise every caller has to remember to do it.